Feature/chequeras

fauno opened the pull request:
son un monton de cambios lo se
- Modelos Cajas y Cuentas se unen en Cajas
- Modelo de Cheques
- Los Movimientos tienen Recibos
- Los Recibos también pueden ser internos
- Las Cajas pueden hacer transferencias, emitir y cobrar cheques

fauno committed:
Agregar Recibo a Movimientos

fauno committed:
Agregar Recibo a Movimientos

fauno committed:
Asociación #32

fauno committed:
Asociación #32

fauno committed:
tratando de arreglar los tests

fauno committed:
tratando de arreglar los tests

fauno committed:
no requerir factura en recibos internos

fauno committed:
no requerir factura en recibos internos

fauno committed:
si se crea un movimiento sin recibo se asume que se esta creando un recibo interno

fauno committed:
si se crea un movimiento sin recibo se asume que se esta creando un recibo interno

fauno committed:
Revert "tratando de arreglar los tests"

This reverts commit 9bec179ead47a201d5f98f392686a2b8684ce526.

Conflicts:
	app/models/movimiento.rb

fauno committed:
Revert "tratando de arreglar los tests"

This reverts commit 9bec179ead47a201d5f98f392686a2b8684ce526.

Conflicts:
	app/models/movimiento.rb

fauno committed:
No validar factura cuando el recibo es interno (esta vez de verdad)

fauno committed:
No validar factura cuando el recibo es interno (esta vez de verdad)

fauno committed:
Si el movimiento no tiene recibo, se cree un recibo interno de monto 0

fauno committed:
Si el movimiento no tiene recibo, se cree un recibo interno de monto 0

fauno committed:
Las transacciones de Caja generan Recibos

burocracia!

fauno committed:
Merge branch 'feature/recibos-movimientos' of github.com:lainventoria/cp into feature/recibos-movimientos

Conflicts:
	app/models/movimiento.rb

fauno committed:
ya se que me faltan los tests

fauno committed:
ya se que me faltan los tests

fauno committed:
Testear vencimiento de los cheques

fauno committed:
Testear vencimiento de los cheques

fauno committed:
Testear en tiempos aleatorios

fauno committed:
Testear en tiempos aleatorios

fauno committed:
Controlador de cheques

fauno committed:
Controlador de cheques

fauno committed:
Testear controlador y arranco con las vistas

fauno committed:
Testear controlador y arranco con las vistas

fauno committed:
Los cheques van a ser mostrados

fauno committed:
Los cheques de terceros no llevan cuenta

fauno committed:
Estas vistas no las vamos a usar

fauno committed:
El scope Cheques.vencidos trae todos los cheques vencidos

fauno committed:
Merge branch 'feature/chequeras' of github.com:lainventoria/cp into feature/chequeras

Conflicts:
	app/controllers/cheques_controller.rb
	app/models/cheque.rb
	config/routes.rb
	test/models/cheques_test.rb

fauno committed:
me quedo un merge extra

fauno committed:
Estas vistas no hay que testearlas

fauno committed:
typo

fauno committed:
Typo

fauno committed:
dummy data para cheques

fauno committed:
Listar cheques

fauno committed:
un poco de variedad en los datos

fauno committed:
Para las vistas

fauno committed:
mejor ser específicos

fauno committed:
no se pluralizan bien las palabras que terminan en e #45

m4tu5 committed:
Nuevos campos en cheque

m4tu5 committed:
gui inicial para cheques

m4tu5 committed:
Merge branch 'feature/chequeras' of github.com:lainventoria/cp into feature/chequeras

fauno committed:
limpieza

m4tu5 committed:
cheques helper

fauno committed:
Merge branch 'feature/chequeras' of github.com:lainventoria/cp into feature/chequeras

m4tu5 committed:
arreglos a mano a los .yml para cheques recibos y facturas

fauno committed:
Merge branch 'feature/chequeras' of github.com:lainventoria/cp into feature/chequeras

fauno committed:
Los cheques también se relacionan con recibos

fauno committed:
Merge branch 'feature/recibos-movimientos' of github.com:lainventoria/cp into feature/chequeras

Conflicts:
	app/models/recibo.rb
	db/schema.rb

fauno committed:
la inflexion ya esta arreglada

fauno committed:
cuando se deposita un cheque se genera un movimiento en el recibo

fauno committed:
el beneficiaio siempre tiene que estar

fauno committed:
Unir Cajas y Cuentas

fauno committed:
Deshacerse de Cuentas, los Cheques pertenecen a Cajas de banco

fauno committed:
remover cuenta_id

fauno committed:
Los cheques tienen estados

fauno committed:
API para depositar cheques de terceros

fauno committed:
En realidad los cheques de terceros primero se depositan y luego se cobran

fauno committed:
Un scope para los Cheques.depositados

fauno committed:
algunos tests fallaban

fauno committed:
Caja.emitir_cheque y notas

fauno committed:
Caja.transferir para movimientos entre cajas

mauriciopasquier commented on app/helpers/cheques_helper.rb:
```diff
@@ -0,0 +1,31 @@
+module ChequesHelper
+
+  def titulo_vista_cheque
+    if :propio?
```
siempre da true

mauriciopasquier commented on app/helpers/cheques_helper.rb:
```diff
@@ -0,0 +1,31 @@
+  end
+
+  def path_a_listado
+    if :propio?
```
siempre da true

fauno replied:
@m4tu5

fauno replied:
@mauriciopasquier como se llama al modelo desde un helper?

mauriciopasquier replied:
On Fri, Mar 07, 2014 at 06:27:06AM -0800, fauno wrote:

> > -      "Detalles de Cheque Propio"
> > -    else
> > -      "Detalles de Cheque de Tercero"
> > -    end
> > -  end 
> >   +
> > -  def titulo_listado_cheques
> > -    if @situacion == 'propios'
> > -      "Listado de Cheques Propios"
> > -    else
> > -      "Listado de Cheques de Terceros"
> > -    end
> > -  end
> >   +
> > -  def path_a_listado
> > -    if :propio?
> 
> @mauriciopasquier como se llama al modelo desde un helper?

se supone que lo tenés en una variable de instancia @modelo. Todas las
variables de instancia que cargues en el controlador se pasan a las vistas y
los helpers usan el mismo contexto

##

fauno replied:
Mauricio Pasquier Juan notifications@github.com writes:

> > @mauriciopasquier como se llama al modelo desde un helper?
> 
> se supone que lo tenés en una variable de instancia @modelo. Todas las
> variables de instancia que cargues en el controlador se pasan a las vistas y
> los helpers usan el mismo contexto

y cuando los queres usar en un elemento de una coleccion conviene que
reciban la instancia como parametro?

## 

http://utopia.partidopirata.com.ar/

mauriciopasquier commented on app/models/caja.rb:
```diff
@@ -19,6 +23,14 @@ class Caja < ActiveRecord::Base
+  end
+
+  def efectivo?
+    :situacion == 'efectivo'
```
qué onda con los dos puntos? Estás usando un literarl `:situacion` no llamando al método.

fauno committed:
Siempre me olvido que el controlador tiene un filtro de parametros

mauriciopasquier commented on app/models/cheque.rb:
```diff
@@ -0,0 +1,81 @@
+  # cheque y se marca como estado = cobrado
+  def cobrar
+    # solo los cheques depositados se pueden cobrar
+    return nil unless :depositado?
```
:

fauno replied:
arreglalo vos y después sigo tu ejemplo

fauno replied:
ahi lo arregle yo

m4tu5 committed:
correccion de error en recibos/form

el error estaba en usar @recibos.factura en recibos nuevos porque
@recibos.factura es nil hasta que se guarda el nuevo recibo

se cambio a @factura, lo que si funciona

fauno committed:
no usar simbolos aca

fauno committed:
testear que no se puedan depositar cheques propios

fauno commented:
listop

fauno committed:
Faltaban cosas en el controlador de cheques

fauno committed:
outdatear comentarios

fauno committed:
Array de monedas usadas en una caja

fauno commented:
@m4tu5 cuando mergeemos esto te podés largar con lo de facturas-recibos-movimientos en la gui...

fauno commented:
y hay que hacer las vistas de cheques, transferencias, etc

mauriciopasquier commented:
cuando mergeemos actualizo lo de las retenciones y hago la interfaz de carga que hablamos el viernes..

fauno commented:
se puede mergear entonces?

fauno commented:
a las 13hs mergeo :)

m4tu5 commented:
@fauno, como sulucionaste de los pasamanos? no te parece mejor ya meterlo de una en este PR? viste la idea de hacer lo de origen_id y destino_id? creo que es lo mas claro.

sino, usar cuenta_id para guardar el recibo de pago cuando sea pasamanos es lo mismo, pero con nombres que generan confusion.

fauno commented:
m4tu5 notifications@github.com writes:

> @fauno, como sulucionaste de los pasamanos? no te parece mejor ya
> meterlo de una en este PR? viste la idea de hacer lo de origen_id y
> destino_id? creo que es lo mas claro.
> 
> sino, usar cuenta_id para guardar el recibo de pago cuando sea
> pasamanos es lo mismo, pero con nombres que generan confusion.

no lo resolvi porque los fines de semana no trabajo B)

me gustaría mergear esto así ya pueden seguir laburando a partir de
estos cambios y después meter el pasamanos en su propia rama

## 

http://endefensadelsl.org

m4tu5 commented on app/models/cheque.rb:
```diff
@@ -0,0 +1,81 @@
+  SITUACIONES = %w(propio terceros)
+  validates_inclusion_of :situacion, in: SITUACIONES
+
+  ESTADOS = %w(chequera depositado cobrado rechazado)
```
faltan el estado y el metodo 'pagar' para los cheques propios.
el metodo tiene que descontar el monto del cheque de la cuenta bancaria asociada, hacer el ajuste al recibo de pago que corresponda y cambiar el estado del cheque a pagado

fauno replied:
m4tu5 notifications@github.com writes:

> faltan el estado y el metodo 'pagar' para los cheques propios.  el
> metodo tiene que descontar el monto del cheque de la cuenta bancaria
> asociada, hacer el ajuste al recibo de pago que corresponda y cambiar
> el estado del cheque a pagado

ok

## 

:{

m4tu5 commented on app/views/cheques/_list.html.erb:
```diff
@@ -0,0 +1,35 @@
+    $(document).ready(function(){
+        $('tr').on('click', function() { window.location = this.id});
+  });
+</script>
```
despues de mergeado, paso esto a assets cuando retoque las vistas

m4tu5 commented on app/views/recibos/_form.html.erb:
```diff
@@ -67,7 +67,6 @@
     <div class="row">
     <h6>Campos Ocultos (debug)<small>
   <% end %>
```
Esto de campos ocultos lo corregi en lo que estoy haciendo para Terceros.
Despues de mergear esto, podemos ver de mergear Terceros como está hasta ahora (pasan los tests!)

m4tu5 commented on db/migrate/20140225012156_add_recibo_to_movimientos.rb:
```diff
@@ -0,0 +1,5 @@
+class AddReciboToMovimientos < ActiveRecord::Migration
+  def change
+    add_column :movimientos, :recibo_id, :integer
+  end
```
perdon con tantas idas y vueltas, creo que el comentario es valido.

no es add_reference en lugar de add_column?

fauno replied:
m4tu5 notifications@github.com writes:

> > @@ -0,0 +1,5 @@
> > +class AddReciboToMovimientos < ActiveRecord::Migration
> > -  def change
> > -    add_column :movimientos, :recibo_id, :integer
> > -  end
> 
> perdon con tantas idas y vueltas, creo que el comentario es valido.
> 
> no es add_reference en lugar de add_column?

eeeh si

## 

http://partidopirata.com.ar

m4tu5 commented:
bueno, ahi estan las cosas que pude detectar

+1 al PR

fauno commented:
ahi puse Cheque.pagar

fauno committed:
Los cheques propios se pueden pagar

fauno committed:
testear un poco mas

fauno committed:
Usar operaciones de Caja para trabajar con Cheques

fauno commented:
no se que onda con los dummy data hubo conflictos porque algunos campos se hicieron binarios (???)

m4tu5 commented:
ni idea ... tal vez sea parte de los cambios que hizo @rata0071 en su momento?

yo borre todo lo que tenia y ahora voy armando data a mano porque sino, es un quilombo

sudoaza commented:
yo regenere los dummy y los edite todos a mano, despues los importe y me
anduvieron joya!

aza

2014-03-10 14:51 GMT-03:00 m4tu5 notifications@github.com:

> ni idea ... tal vez sea parte de los cambios que hizo @rata0071https://github.com/rata0071en su momento?
> 
> yo borre todo lo que tenia y ahora voy armando data a mano porque sino, es
> un quilombo
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/lainventoria/cp/pull/48#issuecomment-37212411
> .
